src/interpreter/crypto.rs

```diff
@@ -1,37 +1,38 @@
 use super::Interpreter;
-use crate::consts::{MEM_MAX_ACCESS_SIZE, VM_MAX_RAM};
-use crate::crypto;
+use crate::consts::{MEM_MAX_ACCESS_SIZE, MIN_VM_MAX_RAM_USIZE_MAX, VM_MAX_RAM};
 use crate::error::RuntimeError;
 
+use crate::arith::{checked_add_word, checked_sub_word};
 use fuel_asm::PanicReason;
-use fuel_crypto::Hasher;
+use fuel_asm::PanicReason::ArithmeticOverflow;
+use fuel_crypto::{Hasher, Message, PublicKey, Signature};
 use fuel_types::{Bytes32, Bytes64, Word};
 
 impl<S> Interpreter<S> {
     pub(crate) fn ecrecover(&mut self, a: Word, b: Word, c: Word) -> Result<(), RuntimeError> {
```

Comment:
I think better to have more descriptive naming.
- 'a' -> `out_ptr`
- 'b' -> `sig_ptr`
- 'c' -> `m_ptr`

Also, instead of `Word`, it may be better to introduce the type as `Ptr` and `PtrMut`. In this case we don't need to add `ptr` suffix=)

Reply:
If we do this (better names are always good), we should do this outside of the function parameters as these letters are mapped to the opcode argument names from the spec.

Reply:
Maybe we also need to do it for opcodes, but I think we can start from `Interpreter`. Because the context of those arguments is important here=) How opcodes transfer them is not so important.

Reply:
created issue https://github.com/FuelLabs/fuel-vm/issues/220 to address this broadly